ui/src/app/preferences/Preferences.tsx

```diff
@@ -1,94 +1,197 @@
 import { Formik } from 'formik';
-import { useMemo } from 'react';
+import { Clock, List, Moon, Sun } from 'lucide-react';
+import { useEffect, useMemo, useRef } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 
 import { Switch } from '~/components/Switch';
 import Select from '~/components/forms/Select';
 
-import { Theme, Timezone } from '~/types/Preferences';
+import { Theme, Timezone, ViewMode } from '~/types/Preferences';
 
+import { useNotification } from '~/data/hooks/notification';
 import { useTimezone } from '~/data/hooks/timezone';
 
 import {
+  resetLastSaved,
+  selectLastSaved,
   selectTheme,
   selectTimezone,
+  selectViewMode,
   themeChanged,
-  timezoneChanged
+  timezoneChanged,
+  viewModeChanged
 } from './preferencesSlice';
 
 export default function Preferences() {
   const timezone = useSelector(selectTimezone);
   const theme = useSelector(selectTheme);
+  const viewMode = useSelector(selectViewMode);
+  const lastSaved = useSelector(selectLastSaved);
 
   const dispatch = useDispatch();
+  const { setNotification } = useNotification();
 
   const initialValues = {
     timezone: timezone,
-    theme: theme
+    theme: theme,
+    viewMode: viewMode
   };
 
   const { inTimezone } = useTimezone();
   const isUTC = useMemo(() => timezone === Timezone.UTC, [timezone]);
 
+  // Debounce timer for notifications
+  const debounceTimerRef = useRef<NodeJS.Timeout | null>(null);
+  const DEBOUNCE_DELAY = 1000; // 1 second
+
+  // Show toast when preferences are saved, with debounce
+  useEffect(() => {
+    if (lastSaved) {
+      // Clear any existing timer
+      if (debounceTimerRef.current) {
+        clearTimeout(debounceTimerRef.current);
+      }
+
+      // Set new timer
+      debounceTimerRef.current = setTimeout(() => {
+        setNotification('Preferences saved', {
+          description: 'Your preferences have been automatically saved.',
+          duration: 2000
+        });
+        dispatch(resetLastSaved());
+        debounceTimerRef.current = null;
+      }, DEBOUNCE_DELAY);
+    }
+
+    // Cleanup timer on unmount
+    return () => {
+      if (debounceTimerRef.current) {
+        clearTimeout(debounceTimerRef.current);
+      }
+    };
+  }, [lastSaved, dispatch, setNotification]);
```

Comment:
Could you please help me to understand why is it needed? Currently if I load the main page and go to Preferences I see the notification without any action.

Reply:
i fixed the initial load issue, the goal is to prevent the dialog from showing up if you save prefs really quickly, like toggling the button on/off very fast. i only want the. notification to show up once in a given time period but im not sure its working as intended